src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -137,3 +146,46 @@ export default {
     });
   }
 } as OptimizerPlugin;
+
+function moveUpCaretValueRule(
+  rule: ExportableCaretValueRule,
+  sd: ExportableProfile | ExportableExtension,
+  knownSiblings: ExportableCaretValueRule[]
+): void {
+  // if the caretPath ends with coding[0], and there are no sibling rules that use indices on coding,
+  // we can move the rule up a level to take advantage of how SUSHI handles rules on CodeableConcepts.
+  if (rule.caretPath.endsWith('coding[0]')) {
+    const basePath = rule.caretPath.replace(/\.coding\[0]$/, '');
+    const siblings = sd.rules.filter(
+      otherRule =>
+        otherRule instanceof ExportableCaretValueRule &&
+        !knownSiblings.includes(otherRule) &&
+        rule !== otherRule &&
+        rule.path === otherRule.path &&
+        otherRule.caretPath.startsWith(`${basePath}.coding[`)
+    );
+    if (siblings.length === 0) {
+      rule.caretPath = basePath;
+    }
+  }
+}
+function moveUpAssignmentRule(
+  rule: ExportableAssignmentRule,
+  instance: ExportableInstance,
+  knownSiblings: ExportableRule[]
+): void {
+  // if the path ends with coding[0], and there are no sibling rules that use indices on coding,
+  // we can move the rule up a level to take advantage of how SUSHI handles rules on CodeableConcepts.
+  if (rule.path.endsWith('coding[0]')) {
+    const basePath = rule.path.replace(/\.coding\[0]$/, '');
+    const siblings = instance.rules.filter(
+      otherRule =>
+        rule !== otherRule &&
```

Comment:
You're not checking if it is an `AssignmentRule`.  Is that because this will only happen in `Instance`s -- and Instances can only have `AssignmentRule` and `InsertRule` (and InsertRule never has a coded value)?

I'm a little torn on this one, but part of me wants the `AssignmentRule` check in case we ever add support for other type rules in instances.  But the other part of me says we'll probably never do that so it's fine.

I'll leave it to you to decide.